publication-commons/src/test/java/no/unit/nva/publication/permission/strategy/ContributorPermissionStrategyTest.java

```diff
@@ -0,0 +1,79 @@
+import no.unit.nva.model.PublicationOperation;
+import no.unit.nva.model.associatedartifacts.AssociatedArtifactList;
+import no.unit.nva.model.role.Role;
+import no.unit.nva.publication.RequestUtil;
+import nva.commons.apigateway.exceptions.UnauthorizedException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.EnumSource.Mode;
+
+class ContributorPermissionStrategyTest extends PublicationPermissionStrategyTest {
+
+    @ParameterizedTest(name = "Should allow verified contributor {0} operation on non-degree resources")
+    @EnumSource(value = PublicationOperation.class, mode = Mode.EXCLUDE, names = {"DELETE", "TERMINATE",
+        "TICKET_PUBLISH"})
+    void shouldAllowVerifiedContributorOnNonDegree(PublicationOperation operation)
+        throws JsonProcessingException, UnauthorizedException {
+
+        var institution = randomUri();
+        var contributor = randomString();
+        var cristinId = randomUri();
+
+        var requestInfo = createUserRequestInfo(contributor, institution, cristinId, randomUri());
+        var publication = createPublicationWithContributor(contributor, cristinId, Role.CREATOR,
+                                                           randomUri(), randomUri());
+        publication.setAssociatedArtifacts(new AssociatedArtifactList());
+
+        var userInstance = RequestUtil.createUserInstanceFromRequest(requestInfo, identityServiceClient);
+
+
+        Assertions.assertTrue(PublicationPermissionStrategy
+                                  .create(publication, userInstance, uriRetriever)
+                                  .allowsAction(operation));
+    }
+
+    @ParameterizedTest(name = "Should deny verified contributor {0} operation on non-degree resources")
+    @EnumSource(value = PublicationOperation.class, mode = Mode.EXCLUDE, names = {"UNPUBLISH", "UPDATE"})
+    void shouldDenyVerifiedContributorOnNonDegree(PublicationOperation operation)
+        throws JsonProcessingException, UnauthorizedException {
+
+        var institution = randomUri();
+        var contributor = randomString();
+        var cristinId = randomUri();
+
+        var requestInfo = createUserRequestInfo(contributor, institution, cristinId, randomUri());
+        var publication = createPublicationWithContributor(contributor, cristinId, Role.CREATOR,
+                                                           randomUri(), randomUri());
+        var userInstance = RequestUtil.createUserInstanceFromRequest(requestInfo, identityServiceClient);
+
+        Assertions.assertFalse(PublicationPermissionStrategy
+                                   .create(publication, userInstance, uriRetriever)
+                                   .allowsAction(operation));
+    }
+
+    @Disabled("Not valid anymore?")
```

Comment:
I believe this one is not valid anymore. Should check with PO, but as I understand, everyone that can edit/update a publication, can also unpublish it (unless the publication has any published files, as mentioned in my comment above). Might be wise to get verification from OP though.

Reply:
ahh yes I have deleted it before, but a lot of the code here was from an old PR so it snuck in again, will remove